### PR TITLE
[std-utils] automatic error logging

### PR DESCRIPTION
**Pull request by tworec**
Seems slow, but it was fun to do it.

Use `cargo test -p ya-market-decentralized --features ya-market-decentralized/market-test-suite --test test_rest_api -- --test-threads 1` to see how it works

### CODE REVIEW

File: utils/std-utils/src/result.rs [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
+        match self {
+            Err(e) => {
+                backtrace::trace(|frame| {
+                    let mut cont = true;
+                    backtrace::resolve_frame(frame, |symbol| {
+                        if let Some(name) = symbol.name().map(|s| s.to_string()) {
+                            if name.starts_with("<ya") {
+                                let name = name.strip_prefix("<").unwrap();
+                                let name = name.split(" as ").next().unwrap();
+                                log::error!("Error at {}: {}", name, &e);
```
Comment (nieznanysprawiciel) [inline utils/std-utils/src/result.rs:18]: Use log `target` instead adding module name to string:
 https://docs.rs/log/0.4.11/log/macro.info.html

**Review by nieznanysprawiciel — CHANGES_REQUESTED** [review]

File: utils/std-utils/src/result.rs (hunk 1 above)
Comment (tworec) [reply utils/std-utils/src/result.rs:18]: noooice, didn't know that option

Comment (tworec) [reply utils/std-utils/src/result.rs:18]: does not work...

Comment (tworec) [reply utils/std-utils/src/result.rs]: Ok, I've found a way to make it work

```
Dan:~/git/yagna log-err (77d88249)$ cargo test -p ya-market-decentralized --features ya-market-decentralized/market-test-suite --test test_rest_api -- --test-threads 1 
   Compiling ya-std-utils v0.1.0 (/Users/tworec/git/yagna/utils/std-utils)
   Compiling ya-market-decentralized v0.1.0 (/Users/tworec/git/yagna/core/market/decentralized)
    Finished test [unoptimized + debuginfo] target(s) in 24.63s
     Running target/debug/deps/test_rest_api-954deebcd40e1c12

running 7 tests
test test_rest_get_agreement ... Running migration 20200515093513
Running migration 20200515093513
ok
test test_rest_get_demands ... Running migration 20200515093513
ok
test test_rest_get_offers ... Running migration 20200515093513
Running migration 20200515093513
ok
test test_rest_get_proposal ... Running migration 20200515093513
Running migration 20200515093513
ok
test test_rest_invalid_subscription_id_should_return_400 ... Running migration 20200515093513
ok
test test_rest_subscribe_unsubscribe_demand ... Running migration 20200515093513
[2020-09-16T12:56:01Z ERROR ya_market_decentralized::rest_api::requestor::unsubscribe] Error occurred: Demand [5742f35929244b318cd8793a0daee06e-49921ef1508224a84bd976f29f197b04d06b07d080831b61b302aab364a7e4cd] not found.
ok
test test_rest_subscribe_unsubscribe_offer ... Running migration 20200515093513
[2020-09-16T12:56:01Z ERROR ya_market_decentralized::rest_api::provider::unsubscribe] Error occurred: Offer [a2f1af7f356d4c9ab6b6d32629f0dd1d-370df3ecca40040727f4699260ef5b52ef21fba3aeba422dc24da553a26a9f6d] already unsubscribed.
ok

test result: ok. 7 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


```

**Review by nieznanysprawiciel — APPROVED** [review]

**Pull request merged**